Race condition?

Hey,

My linter is giving me

```
...
[2020-12-16T10:53:16Z] /root/.cache/bazel/_bazel_root/26b87eb8c508ba6300e65749e6032969/sandbox/processwrapper-sandbox/2628/execroot/__main__/external/com_github_bitly_go_hostpool/epsilon_greedy.go:52:27: literal copies lock value from *stdHP: github.com/bitly/go-hostpool.standardHostPool
```

And lo and behold, https://github.com/bitly/go-hostpool/blob/5d3b4dc6ed4701a761ee884fd803b0a0879744f1/epsilon_greedy.go#L52 _does_ make a copy of `sync.Mutex`.

Fix can be found [here](https://github.com/JensRantil/go-hostpool/commit/b86b322de56434aa7f569ce1dcb30783e91544e6).

--- a/epsilon_greedy.go
+++ b/epsilon_greedy.go
@@ -20,7 +20,7 @@ func (r *epsilonHostPoolResponse) Mark(err error) {
 }
 
 type epsilonGreedyHostPool struct {
-	standardHostPool               // TODO - would be nifty if we could embed HostPool and Locker interfaces
+	*standardHostPool              // TODO - would be nifty if we could embed HostPool and Locker interfaces
 	epsilon                float32 // this is our exploration factor
 	decayDuration          time.Duration
 	EpsilonValueCalculator // embed the epsilonValueCalculator
@@ -49,7 +49,7 @@ func NewEpsilonGreedy(hosts []string, decayDuration time.Duration, calc EpsilonV
 	}
 	stdHP := New(hosts).(*standardHostPool)
 	p := &epsilonGreedyHostPool{
-		standardHostPool:       *stdHP,
+		standardHostPool:       stdHP,
 		epsilon:                float32(initialEpsilon),
 		decayDuration:          decayDuration,
 		EpsilonValueCalculator: calc,

